Camera Stream No Longer Works

### The problem

I'm assuming when the integration polls the api for the sensor data it is assuming that the camera stream is on the same port at the Octoprint instance. For those not running their instances on a Raspberry pi this typically isn't the case. I'm running MJPG streamer on ubuntu server.  The stream works great in Octoprint, but home assistant fails to retrieve the correct port. My octoprint instance runs on :5000 while my camera stream is on :8080, but the set up wizard doesn't get that information. So, the camera entity is stuck on state Idle while the feed says, "Preview of the octoprint camera". 

Using a generic camera will show the stream snapshot, but fails to render the live stream.

### What version of Home Assistant Core has the issue?

2023.1.0

### What was the last working version of Home Assistant Core?

2022.12

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Octoprint

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/octoprint/

### Diagnostics information

[home-assistant_octoprint_2023-01-06T03-37-51.433Z.log](https://github.com/home-assistant/core/files/10357153/home-assistant_octoprint_2023-01-06T03-37-51.433Z.log)


### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
2023-01-05 21:24:52.437 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new camera.octoprint entity: camera.octoprint_camera
2023-01-05 21:24:53.192 WARNING (MainThread) [homeassistant.components.cloud.google_config] Error reporting state - 5: Requested entity was not found.
2023-01-05 21:25:02.138 ERROR (MainThread) [frontend.js.latest.202301040] http://homeassistant.local:8123/frontend_latest/app.c7254a29.js:850:0 Error: Failed to execute 'define' on 'CustomElementRegistry': the name "button-card-action-handler" has already been used with this registry
2023-01-05 21:25:14.312 INFO (MainThread) [homeassistant.components.websocket_api.http.connection] [546872166240] Connection closed by client
```


### Additional information

I have deleted the integration several times as well as multiple reboots and defining the camera manually in camera.yaml
If this is an error on my end I apologize in advance, but my camera stream was working for days up until now. If it is an error on my end, anything ideas would be greatly appreciated.

--- a/homeassistant/components/octoprint/manifest.json
+++ b/homeassistant/components/octoprint/manifest.json
@@ -6,7 +6,7 @@
   "documentation": "https://www.home-assistant.io/integrations/octoprint",
   "iot_class": "local_polling",
   "loggers": ["pyoctoprintapi"],
-  "requirements": ["pyoctoprintapi==0.1.9"],
+  "requirements": ["pyoctoprintapi==0.1.11"],
   "ssdp": [
     {
       "manufacturer": "The OctoPrint Project",

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -1834,7 +1834,7 @@ pynzbgetapi==0.2.0
 pyobihai==1.3.2
 
 # homeassistant.components.octoprint
-pyoctoprintapi==0.1.9
+pyoctoprintapi==0.1.11
 
 # homeassistant.components.ombi
 pyombi==0.1.10

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -1323,7 +1323,7 @@ pynx584==0.5
 pynzbgetapi==0.2.0
 
 # homeassistant.components.octoprint
-pyoctoprintapi==0.1.9
+pyoctoprintapi==0.1.11
 
 # homeassistant.components.openuv
 pyopenuv==2023.02.0

